.github/workflows/build.yml

```diff
@@ -0,0 +1,271 @@
+              {"summary": "A Maintainer needs to apply the `merge-ready` label, which will trigger a fully featured build."}
+ 
+  
+  # Our full integration job, which will build for a matrix out of our supported
+  # java versions, operating systems, modular or not, and of cource junit versions.
+  # This build is executed on all tags and the default branch. Furthermore we will
+  # also execute this for pullrequests with the label `merge-ready`.
+  full-featured:
+    if: (contains(github.event.pull_request.labels.*.name, 'merge-ready') || !github.event.pull_request)
+    needs: basic
+    runs-on: ${{ matrix.os }}-latest
+    timeout-minutes: 15
+    strategy:
+      matrix:
+        java: [ 8, 11, 16 ]
+        junit-version: [ '5.7.0', '5.7.1', '5.8.0-M1' ]
+        modular: [true, false]
+        os: [ubuntu, macos, windows]
+        exclude:
+          - java: 8
+            modular: true
+    name: with Java ${{ matrix.java }}, JUnit ${{ matrix.junit-version }}, Modular ${{ matrix.modular }} on ${{ matrix.os }}
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v2
+      - name: Set up java
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+      - name: Cache Gradle
+        uses: actions/cache@v2
+        with:
+          path: ~/.gradle/caches/
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: ${{ runner.os }}-gradle-
+      - name: Gradle build
+        uses: eskatos/gradle-command-action@v1
+        with:
+          arguments: --refresh-dependencies -PmodularBuild=${{ matrix.modular }} -PjunitVersion=${{ matrix.junit-version }} --stacktrace --scan clean build
+
+  # We want to be up to date and know issues with future java versions as soon as possible.
+  # Furthermore we also would love to see our build working with the latest gradle version.
+  # As those builds might fail, they are allowed to fail, and should not prevent merges.
+  experimental:
+    if: (contains(github.event.pull_request.labels.*.name, 'merge-ready') || !github.event.pull_request)
+    needs: full-featured
+    runs-on: ${{ matrix.os }}-latest
+    continue-on-error: true
+    timeout-minutes: 15
+    strategy:
+      matrix:
+        junit-version: [ '5.7.0', '5.7.1', '5.8.0-M1' ]
+        modular: [true, false]
+        os: [ubuntu, macos, windows]
+    name: Experimental build with java 17 and Gradle Release Candidate
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v2
+      - uses: sormuras/download-jdk@v1
+        with:
+          feature: ${{ env.EXPERIMENTAL_JAVA }}
+      - uses: actions/setup-java@v2
+        with:
+          java-version: ${{ env.JDK_VERSION }}
+          distribution: jdkfile
+          jdkFile: ${{ env.JDK_FILE }}
+      - name: Prepare JDK_EXPERIMENTAL env var
+        shell: bash
+        run: echo "JDK_EXPERIMENTAL=$JAVA_HOME" >> $GITHUB_ENV
+      - uses: actions/setup-java@v2
```

Comment:
Hmmm, why this additional Java setup with v16 and AdoptOpenJDK?

Reply:
gradle is not working with ea builds - it will fail, so we start gradle with a suported java version, but execute the build on the experimental version

Reply:
Would a comment make sense then?

Reply:
I added a comment.